### PR TITLE
Feat: Specific Milestone Funding Ability

### LINKED ISSUE #176: Add specific milestone funding ability
Module that integrates directly into the milestoneManager
Fund a specific Milestone

### PR DESCRIPTION
**Pull request by FHieser**
See #176

### CODE REVIEW

File: src/modules/milestoneSubModules/ISpecificFundingManager.sol [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    //--------------------------------------------------------------------------
+    // Functions
+
+    //----------------------------------
+    // View Functions
+
+    /// @notice Returns the amount of tokens all funders deposited for a milestone with id 'milestoneId'.
+    /// @param milestoneId : The id of the milestone that the funders deposited to
+    /// @return amount : amount of tokens the funders deposited
+    function getFunderAmountForMilestoneId(uint milestoneId)
```
Comment (SchneiderNejc) [inline src/modules/milestoneSubModules/ISpecificFundingManager.sol:86]: rename to getFundedAmountForMilestoneId

File: src/modules/milestoneSubModules/SpecificFundingManager.sol [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+    {
+        milestoneIdToFundingAddresses[milestoneId].fundingCollected = true;
+
+        address[] memory funders =
+            milestoneIdToFundingAddresses[milestoneId].funders;
+
+        uint length = funders.length;
+
+        //If we dont have any funders stop the collecting
+        if (length == 0) {
```
Comment (SchneiderNejc) [inline src/modules/milestoneSubModules/SpecificFundingManager.sol:242]: validAmount modifier is checked to not be zero, therefore it has to have some funders ?
making if section 242 - 245 redundant.

**Review by SchneiderNejc — APPROVED** [review]
I'd like to go over collectFunding() function and maybe change some naming conventions.

**Comment by FHieser** [comment]
As milestonemanager was put on ice i will close this too

**Pull request closed without merging**